engine/grid.py

```diff
@@ -300,73 +307,164 @@ def find_border_nodes(self):
         self.leftmost_node_pos = leftmost_node_pos
+        self.rightmost_node = rightmost_node
+        self.rightmost_node_pos = rightmost_node_pos
 
-    # Return bottom most node that is activated in the right column
-
+    
+    ##Return bottom most node that is activated in the right column
     def bottom_rightmost_node(self, pos):
-
         # node_info: (node, row, col)
         candidate_nodes = [node_info for node_info in self.border_nodes if node_info[2] == pos[1]+1]
         if (candidate_nodes == []):
             return None
         else:
             node_info = min(candidate_nodes,key=lambda node_info: node_info[1])
             return (node_info[1],node_info[2])
+    
+    def bottom_leftmost_node(self, pos):
+        # node_info: (node, row, col)
+        candidate_nodes = [node_info for node_info in self.border_nodes if node_info[2] == pos[1]+1]
+        if (candidate_nodes == []):
+            return None
+        else:
+            node_info = max(candidate_nodes,key=lambda node_info: node_info[1])
+            return (node_info[1],node_info[2])
 
+    
 
-    # Given border and active nodes, compute lawnmower traversal
-
+        
+    ##Given border and active nodes, compute lawnmower traversal
     def get_all_lawnmower_waypoints_adjustable(self):
         class WaypointPhase(Enum):
             DOWN = 1
             TERMINATE = 2
-        rows = self.nodes.shape[0]
+        class Direction(Enum):
+            RIGHT = 1
+            LEFT = 2
+
+        class Orientation(Enum):
+            CW = 1
+            CCW = 2
+
+        def plot_circle(start_pos, end_pos, center, orientation):
+            """
+            Returns a circle of nodes starting from the start_pos, going at orientation orientation, and ending at the
+            end_pos with center center.
+
+            Arguments:
+                start_pos: float tuple representing the starting point of the circle being plotted
+                end_pos: float tuple representing the ending point of the circle being plotted
+                center: float tuple representing the center point of the circle being plotted
+                orientation: orientation that the nodes are being plotted from, starting with the start_pos and ending
+                    at the end_pos
+            """
+            r = math.hypot(float(start_pos[0]) - center[0], float(start_pos[1]) - center[1])
+            theta_init = math.atan2(start_pos[1]-center[1], start_pos[0]-center[0])
+            theta_end = math.atan2(end_pos[1]-center[1], end_pos[0]-center[0])
+            circle_plt = []
+            if theta_end < theta_init:
+                theta_end = theta_end + 2*math.pi
+            if orientation == Orientation.CCW:
+                theta = theta_init
+                while theta < theta_end:
+                    circle_plt.append((r*math.cos(theta)+center[0], r*math.sin(theta)+center[1]))
+                    theta = theta + math.pi/12
```

Comment:
can be made into param with default value math.pi/12 to give user customizability to how smooth the curve is

Reply:
I made it into a param, but I am not sure if the naming and description are accurate. does `smoothness: int representing how subdivided the angle will be when plotting the turning arch. The 
                greater the value, the smoother the curve will be. Default value = 12.` sound right?

Reply:
i think it'll be better to use something like theta_step and have the user pass in pi/12 instead of 12

Reply:
That makes sense, I fixed it